parliament_challenge.py

```diff
@@ -0,0 +1,149 @@
+    """
+
+    speeches_list = []
+    for s in speeches:
+        speechesDict = {
+                "anforande_id": s.get("anforande_id"),
+                "dok_datum": s.get("dok_datum"),
+                "parti": s.get("parti"), 
+                "avsnittsrubrik": s.get("avsnittsrubrik"),
+                "links": [{ 
+                    "rel": "speech",
+                    "href": s.get("protokoll_url_www")
+                        }],
+                "intressent_id": s.get("intressent_id")
+                }
+        speeches_list.append(speechesDict)
+    return speeches_list
+
+
+def filter_member_dict(member):
+    """
+    Filtering member data for relevant keys.
+    Relevant keys are tilltalsnamn, valkrets,
+    bild_url_192, uppgift and intressent_id.
+    """
+
+    if member["personuppgift"] is None:
+        email = None
+    else:
+        for e in member["personuppgift"]["uppgift"]:
+            if e["kod"] == "Officiell e-postadress":
+                uppgift = e["uppgift"]
+                email = uppgift[0]
+
+    membersDict = {
+        "tilltalsnamn": member.get("tilltalsnamn"),
+        "valkrets": member.get("valkrets"),
+        "uppgift": email,
+        "bild_url_192": member.get("bild_url_192"),
+        "intressent_id": member.get("intressent_id")
+    }
+    return membersDict
+
+
+def get_member_data(intressent_id):
+    """
+    Get a member data from the member's api
+    and filter for relevant keys.
+    """
+
+    domain = 'http://data.riksdagen.se'
+    members = 'personlista'
+    format_type = 'json'
+    response = requests.get('{}/{}/?iid={}&utformat={}'.format(domain, members, intressent_id, format_type))
+    if response.status_code == 200:
+        data = response.json()
+        # One member data being returned at a time.
+        if data["personlista"]["@hits"] == "1":
+            member = data["personlista"]["person"]
+            filtered_member = filter_member_dict(member)
+            return filtered_member, response.status_code
+        # In case a member detail is non-existent,
+        # returning an empty dict to indicate that 
+        # the detail does not exist in the member data.
+        elif data["personlista"]["@hits"] == "0":
+            nullMemberDict = {}
+            return nullMemberDict, response.status_code
+    else:
+        return {}, response.status_code
+
+
+def get_speeches_data(anftyp, size):
+    """
+    Get speeches data from speech's api
+    and filter for relevant keys.
+    For each speech we query the member api 
+    to get the member data and 
+    then merge it with the speech data.
+    """
+
+    domain = 'http://data.riksdagen.se'
+    speeches = 'anforandelista'
+    format_type = 'json'
+    response = requests.get('{}/{}/?anftyp={}&sz={}&utformat={}'.format(domain, speeches, anftyp, size, format_type))
+    if response.status_code == 200:
+        data = response.json()
+        # When a client requests for more than one speech.
```

Comment:
This comment is wrong right? The client doesn't request for more than one speech? The speech api  server returned more than one speech.